includes/managers/icons/migrations.php

```diff
@@ -0,0 +1,298 @@
+			 * @since 3.19.0
+			 * @param bool $migration_required
+			 */
+			$migration_required = apply_filters( 'elementor/icons_manager/migration_required', $migration_required );
+		}
+
+		return $migration_required;
+	}
+
+	public function register_ajax_actions( Ajax $ajax ) {
+		$ajax->register_ajax_action( 'icon_manager_migrate', [ $this, 'ajax_upgrade_to_current_version' ] );
+	}
+
+	/**
+	 * Ajax Upgrade to the current version of Font Awesome
+	 */
+	public function ajax_upgrade_to_current_version() {
+		if ( ! current_user_can( 'manage_options' ) ) {
+			throw new \Exception( 'Permission denied' );
+		}
+
+		$this->run_migration();
+		self::update_migration_required_flags();
+
+		$success_title = sprintf(
+			/* translators: %s is the version number. */
+			esc_html__( 'You\'re up-to-date with Font Awesome %s!', 'elementor' ),
+			Icons_Manager::get_current_fa_version()
+		);
+
+		return [
+			'title' => $success_title,
+			'message' => esc_html__( 'Elevate your designs with these new and updated icons.', 'elementor' ),
+		];
+	}
+
+	/**
+	 * Run the migration process.
+	 */
+	private function run_migration() {
+		$custom_tasks = Plugin::instance()->custom_tasks;
+
+		$custom_tasks->add_tasks_requested_to_run( [ [ 'Elementor\Core\Upgrade\Custom_Tasks', 'migrate_fa_icon_values' ] ] );
+		$updater = $custom_tasks->get_task_runner();
+
+		$callbacks = $custom_tasks->get_tasks_requested_to_run();
+		$did_tasks = false;
+
+		if ( ! empty( $callbacks ) ) {
+			$updater->handle_immediately( $callbacks );
+			$did_tasks = true;
+		}
+
+		$custom_tasks->on_runner_complete( $did_tasks );
+	}
+
+	/**
+	 * Remove the update flag and update the latest migration flag.
+	 */
+	public static function update_migration_required_flags() {
+		delete_option( 'elementor_' . self::NEEDS_UPDATE_OPTION );
+		update_option( 'elementor_' . self::LATEST_MIGRATION_OPTION, self::LATEST_MIGRATION_REQUIRED_VERSION );
+	}
+
+	/**
+	 * Migrate the icon values.
+	 * This function is called from the Custom_Tasks class.
+	 *
+	 * @param $updater
+	 * @return void
+	 */
+	public static function migrate_icon_values( $updater ) {
+		$changes = [ [ 'callback' => [ 'Elementor\Icons_Manager\Migrations', 'migrate_icon_value_names' ] ] ];
+
+		$should_run_again = true;
+		while ( $should_run_again ) {
+			$should_run_again = Upgrade_Utils::_update_widget_settings( '*', $updater, $changes );
+		}
+	}
+
+	/**
+	 * Migrate the icon value names.
+	 * Used during the iteration of the widgets in the full migration process.
+	 *
+	 * @param $element
+	 * @param $args
+	 *
+	 * @return array|mixed
+	 */
+	public static function migrate_icon_value_names( $element, $args ) {
+		if ( empty( $element['widgetType'] ) || ! in_array( $args['widget_id'], [ '*', $element['widgetType'] ] ) ) {
+			return $element;
+		}
+
+		foreach ( $element['settings'] as $key => $value ) {
+			if ( ! empty( $value['value'] ) && is_string( $value['value'] ) && str_contains( $value['value'], ' fa-' ) ) {
```

Comment:
Maybe move the long condition to a variable to be more clear about what are you checking.